### PR TITLE
backport: bitcoin#17509, #17579, #18451, #18504, #18581, #18645, #18646, #18667, #18673, #18682

### PR DESCRIPTION
**Pull request by knst**
## Issue being fixed or feature implemented
Various backports from v21

## What was done?

- bitcoin/bitcoin#18504
- bitcoin/bitcoin#18673
- bitcoin/bitcoin#18581
- bitcoin/bitcoin#18646
- bitcoin/bitcoin#18451
- bitcoin/bitcoin#18645
- bitcoin/bitcoin#18667
- bitcoin/bitcoin#18682
- bitcoin/bitcoin#17509
- bitcoin/bitcoin#17579

## How Has This Been Tested?
Run unit/functional tests

## Breaking Changes
N/A

## Checklist:
- [x] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have added or updated relevant unit/integration/functional/e2e tests
- [ ] I have made corresponding changes to the documentation
- [x] I have assigned this pull request to a milestone

### CODE REVIEW

File: src/wallet/rpcwallet.cpp [hunk 1]
```diff
@@ -675,37 +721,7 @@ static UniValue getreceivedbyaddress(const JSONRPCRequest& request)
...
-    CScript scriptPubKey = GetScriptForDestination(dest);
-    if (!pwallet->IsMine(scriptPubKey)) {
-        throw JSONRPCError(RPC_WALLET_ERROR, "Address not found in wallet");
-    }
-
-    // Minimum confirmations
-    int nMinDepth = 1;
-    if (!request.params[1].isNull())
-        nMinDepth = request.params[1].get_int();
-    bool fAddLocked = (!request.params[2].isNull() && request.params[2].get_bool());
```
Comment (UdjinM6) [inline src/wallet/rpcwallet.cpp:692]: 17579: missing `fAddLocked` and its usage in `GetReceived`

File: src/qt/sendcoinsdialog.cpp [hunk 2]
```diff
@@ -293,37 +290,29 @@ void SendCoinsDialog::on_sendButton_clicked()
...
         }
-        send(recipients);
-        return;
     }
-    // already unlocked or not encrypted at all
-    send(recipients);
-}
 
-void SendCoinsDialog::send(QList<SendCoinsRecipient> recipients)
-{
```
Comment (UdjinM6) [inline src/qt/sendcoinsdialog.cpp:306]: 17509: this should be changed to 
```
        return send(recipients);
    }
    // already unlocked or not encrypted at all
    return send(recipients);
}

bool SendCoinsDialog::send(QList<SendCoinsRecipient> recipients)
{
```
, not dropped. `UnlockContext` is important.

File: src/qt/sendcoinsdialog.h [hunk 3]
```diff
@@ -64,8 +64,8 @@ public Q_SLOTS:
     ClientModel *clientModel;
     WalletModel *model;
     std::unique_ptr<CCoinControl> m_coin_control;
+    std::unique_ptr<WalletModelTransaction> m_current_transaction;
     bool fNewRecipientAllowed;
-    void send(QList<SendCoinsRecipient> recipients);
```
Comment (UdjinM6) [inline src/qt/sendcoinsdialog.h:68]: 17509: should be 
```
    bool send(QList<SendCoinsRecipient> recipients);
```

**Review by UdjinM6 — CHANGES_REQUESTED** [review]

**Review by UdjinM6 — DISMISSED** [review]
utACK

**Comment by knst** [comment]
> @[knst](https://github.com/knst) knst [force-pushed](https://github.com/dashpay/dash/compare/b36f59243ac8072976947a1e87a94a6e07b9795e..fc2a86bee80e45bca9593e9cc0024ff47565d9b9) the bp-v21-p1 branch from[ b36f592 ](https://github.com/dashpay/dash/commit/b36f59243ac8072976947a1e87a94a6e07b9795e)to fc2a86b

rebased on top of develop

**Review by UdjinM6 — APPROVED** [review]
utACK

**Review by PastaPastaPasta — APPROVED** [review]
utACK for merging via merge commit

**Pull request merged**